### PR TITLE
v1.5.1 release candidate

### LINKED ISSUE #154: UX: Add type discovery and casting with proper overload call to all the RollbarLogger's methods that accept object as their first argument.
This will be more client-code-friendly improvement, so that the clients do not have to remember to cast the objects to more specific types to get benefit of more detailed logging. 
It relates to #153

### LINKED ISSUE #155: Samples: Update Xamarin sample's dependencies

### LINKED ISSUE #156: Performance: reuse the same HttpClient instance whenever possible
#138

### LINKED ISSUE #157: Refactoring: extract deployment related functionality of RollbarClient into RollbarDeployClient

### LINKED ISSUE #158: Implement proper hosting OS detection under .Net Core

### LINKED ISSUE #160: Capture CPU architecture as a new data field within Client/Server elements of the payload
make it free formed string for now...

### PR DESCRIPTION
**Pull request by akornich**
- resolve #160: Capture CPU architecture as a new data field within Client/Server elements of the payload
      - resolve #158: Implement proper hosting OS detection under .Net Core
      - resolve #157: Refactoring: extract deployment related functionality of RollbarClient into RollbarDeployClient
      - resolve #156: Performance: reuse the same HttpClient instance whenever possible
      - resolve #155: Samples: Update Xamarin sample's dependencies
      - resolve #154: UX: Add type discovery and casting with proper overload call to all the RollbarLogger's methods that accept object as their first argument

### CODE REVIEW

File: Rollbar/Deploys/RollbarDeployClient.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+        public RollbarDeployClient(RollbarConfig config, HttpClient httpClient = null)
+        {
+            Assumption.AssertNotNull(config, nameof(config));
+
+            this._config = config;
+
+            this._httpClient = httpClient;
+
+            //var header = new MediaTypeWithQualityHeaderValue("application/json");
```
Comment (rokob) [inline Rollbar/Deploys/RollbarDeployClient.cs]: We can get rid of these commented out lines.

File: Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs [hunk 2]
```diff
@@ -51,5 +52,31 @@ public static string[] GetAssemblyTargetFrameworks(Assembly theAssembly)
 
             return targetFrameworks;
         }
+
+        /// <summary>
+        /// Gets the OS description.
+        /// </summary>
+        /// <returns></returns>
+        public static string GetOSDescription()
```
Comment (rokob) [inline Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: This is being used where just plain "windows" was used before. From reading the docs this string will likely have more information and not just be "windows". I think the backend will be able to handle that and still show the right OS logo but I'm not sure. Something similar broke PHP language detection when we enriched the language value. As long as the items being sent still look the same in the UI then everything is okay.

Comment (akornich) [reply Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: Looks like the platform is just presented as a string:
image
and with the code changes it looks like this now:
image
Is there any other place when a logo is presented instead?

File: Rollbar/Deploys/RollbarDeployClient.cs (hunk 1 above)
Comment (akornich) [reply Rollbar/Deploys/RollbarDeployClient.cs]: done. thanks!

File: Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs (hunk 2 above)
Comment (rokob) [reply Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: In the list of items there are logos sometimes depending on the platform and language. I looked into it and I was wrong, this is okay.

Comment (rokob) [reply Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: In the list of items there are logos sometimes depending on the platform and language. I looked into it and I was wrong, this is okay.

Comment (akornich) [reply Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: @rokob , thanks for double checking it. i'll be merging the PR in a few minutes.

Comment (akornich) [reply Rollbar/Diagnostics/RuntimeEnvironmentUtility.cs]: @rokob, btw, it looks like the items with request element populated have OS logos:
image
and the top ones with today's date already use new more detailed OS name/version format. Looks like it works...

**Pull request merged**